academy/ibc-dev/ibcapp_steps.md

```diff
@@ -0,0 +1,425 @@
+
+Ignite CLI implements this in the file `x/leaderboard/module_ibc.go`.
+
+//TODO: add collapsible window??
+
+Additionally, in the `module.go` file, add the following line (and the corresponding import):
+
+```diff
+var (
+    _ module.AppModule      = AppModule{}
+    _ module.AppModuleBasic = AppModuleBasic{}
+    // Add this line
++    _ porttypes.IBCModule   = IBCModule{}
+)
+```
+
+#### Channel handshake version negotiation
+
+Application modules are expected to verify versioning used during the channel handshake procedure.
+
+- `OnChanOpenInit` will verify that the relayer-chosen parameters
+  are valid and perform any custom `INIT` logic.
+  It may return an error if the chosen parameters are invalid
+  in which case the handshake is aborted.
+  If the provided version string is non-empty, `OnChanOpenInit` should return
+  the version string if valid or an error if the provided version is invalid.
+  **If the version string is empty, `OnChanOpenInit` is expected to
+  return a default version string representing the version(s)
+  it supports.**
+  If there is no default version string for the application,
+  it should return an error if the provided version is an empty string.
+- `OnChanOpenTry` will verify the relayer-chosen parameters along with the
+  counterparty-chosen version string and perform custom `TRY` logic.
+  If the relayer-chosen parameters
+  are invalid, the callback must return an error to abort the handshake.
+  If the counterparty-chosen version is not compatible with this module's
+  supported versions, the callback must return an error to abort the handshake.
+  If the versions are compatible, the try callback must select the final version
+  string and return it to core IBC.
+  `OnChanOpenTry` may also perform custom initialization logic.
+- `OnChanOpenAck` will error if the counterparty selected version string
+  is invalid and abort the handshake. It may also perform custom ACK logic.
+
+#### Packet callbacks
+
+The general application packet flow was discussed in [a previous section](https://tutorials.cosmos.network/academy/4-ibc/channels.html#application-packet-flow). 
+
+As a refresher, let's take a look at the diagram: 
+![packet flow](./images/packetflow.png)
+
+We can now identify the packet callbacks in the packet flow by investigating the `IBCModule` interface.
+
+##### Sending packets
+
+Modules **do not send packets through callbacks**, since the modules initiate the action of sending packets to the IBC module, as opposed to other parts of the packet flow where messages sent to the IBC module must trigger execution on the port-bound module through the use of callbacks. Thus, to send a packet a module simply needs to call `SendPacket` on the `IBCChannelKeeper`.
+
+::: warning
+In order to prevent modules from sending packets on channels they do not own, IBC expects modules to pass in the correct channel capability for the packet's source channel.
```

Comment:
maybe link to this doc about ocaps in the sdk at large? not sure if its more confusing tho. https://docs.cosmos.network/v0.44/core/ocap.html